Pipes - Angela - solar_system.rb

```diff
@@ -0,0 +1,377 @@
+########################################
+# Wave 3
+########################################
+
+# Goals
+  # 1. user can select a planet
+  # 2. user can view info about planet
+  # 3. user can add their own planet
+
+
+# Wave three is partially complete because it took me some time to understand the fundamentalls.
+# please provide some good feedback on my existing work.
+
+class Planet
+  # Line below is an attribute reader with all attributes in one method.
+  attr_reader :name, :volume, :solid, :diameter, :orbit_time
+
+
+  def initialize(name, volume, diameter, solid, orbit_time)
+    @name = name
+    @volume = volume
+    @solid = solid
+    @diameter = diameter
+    @orbit_time = orbit_time
+  end
+
+  def planet_output
+    planet_characteristics_ouput = "
+    Your planet's name = #{@name}.
+    Your planet's volume = #{@volume}.
+    Is it solid? #{@solid}.
+    Its diameter is #{@diameter}.
+    It takes #{@orbit_time} to orbit its sun."
+    return planet_characteristics_ouput
+  end
+
+  def planets_name
+    list_planet_names = "#{@name}\n"
+
+    return list_planet_names
+  end
+
+  def planet_detail
+    planet_details = "
+    #{@name}'s volume = #{@volume}.
+    Is it solid? #{@solid}.
+    #{@name} diameter is #{@diameter}.
+    It takes #{@orbit_time} to orbit its sun."
+    return planet_details
+  end
+
+  def name_of_planet
+    return @name
+  end
```

Comment:
Since you've got `attr_reader :name` up top, you don't really need this method. Instead of saying `planet.name_of_planet`, a caller could just use `planet.name`.